scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -111,15 +111,14 @@ private void ensureRequestIsCurrent(final ScheduleRequest request) throws NoSuch
    * @return a problem specification ready for passing to the scheduler
    */
   private Problem createProblem(final PlanMetadata planMetadata) {
-    final var mission = loadMissionModel(planMetadata);
-    final var problem = new Problem(mission);
-
+    final var missionModel = loadMissionModel(planMetadata);
+    final var problem = new Problem(missionModel, planMetadata.horizon());
     //seed the problem with the initial plan contents
-    problem.setInitialPlan(loadInitialPlan(planMetadata, mission));
+    problem.setInitialPlan(loadInitialPlan(planMetadata, problem));
```

Comment:
Quick question:  `problem` is confusing to me. Are problems errors found in the plan or the constraints/rules the plan needs to follow?

The documentation: ` @param problem the problem that the plan adheres to`

Reply:
Maybe this sentence is wrong, that's what you mean? A plan adheres to a mission model but not to a problem. A problem might contain global constraints that need to be enforced during scheduling. And rules/goals that scheduling must try to achieve.

Reply:
Oh, I get it now. `Problems` in this context are the thing the scheduler is trying to solve like a math problem. I was reading it as errors or 'here are problems/error with the plan'